compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -741,6 +816,9 @@ LogicalResult lowerToAIE(ModuleOp moduleOp) {
       static_cast<uint32_t>(device.value()));
 
   auto funcRes = moduleOp.walk([&](func::FuncOp funcOp) {
+    if (funcOp.isPrivate()) {
```

Comment:
What do you think about creating the new functions here and then looking them up later?

Reply:
For creating a new declaration I'd need to know the updated operand types - which at this point I do not.

But since the operands would ALWAYS be `memref<elemType, 1>, index` - I can create them here by traversing each individual argument and creating `memref<elemType, 1>` for `memref` args, while letting the `index` args just be.
Followed by mapping the old one to the new.

But I didn't go for this because :-
1. The updated operand/arg types should just be inferred by the def-chain being updated.
    Eg: Updated `memref.extract_strided_metadata` return types -> `func.call` operand -> func def arguments.
2. Also, I believe the function declaration should be created within `aie.device` - which I cannot do in this step as for each `FuncOp` (public) we are creating a new `aie.device`.

Reply:
Ok, thanks, makes sense.